MapboxNavigation/Cache.swift

```diff
@@ -109,22 +109,22 @@ internal class FileCache {
         }
     }
 
-    private func cachePathWithKey(_ key: String) -> String {
+    func cachePathWithKey(_ key: String) -> String {
         let cacheKey = cacheKeyForKey(key)
         return cacheURLWithKey(cacheKey).absoluteString
     }
 
-    private func cacheURLWithKey(_ key: String) -> URL {
+    func cacheURLWithKey(_ key: String) -> URL {
         let cacheKey = cacheKeyForKey(key)
         return diskCacheURL.appendingPathComponent(cacheKey)
     }
 
-    private func cacheKeyForKey(_ key: String) -> String {
+    func cacheKeyForKey(_ key: String) -> String {
         if let keyAsURL = URL(string: key) {
-            return String.init(keyAsURL.lastPathComponent.hashValue)
+            return MD5(keyAsURL.lastPathComponent)
         }
 
-        return String.init(key.hashValue)
+        return MD5(key)
```

Comment:
This single line is the actual fix. As pointed out by @akitchen, there are some shortcuts String.hashValue takes some shortcuts and things in the middle of the string may not impact the hash.